chrome-extension/content.js

```diff
@@ -0,0 +1,109 @@
+document.body.appendChild(sidebar);
+
+//Inject invisible highlight icon onto page
+const iconButton = document.createElement('button');
+iconButton.classList.add('injection-icon');
+iconButton.innerHTML = 'Save';
+//TODO: CHANGE BUTTON TO AN ACTUAL ICON!!
+
+document.body.appendChild(iconButton);
+
+//Keep track of last used highlight
+let lastContent;
+
+//Show extension icon when selecting text on page
+document.onclick = (event) => {
+  if (iconButton.style.display === 'block') {
+    iconButton.style.display = 'none';
+  }
+  const content = window.getSelection().toString();
+  //TODO: Trim Text
+  if (content && content !== lastContent) {
+    showIcon(event, content);
+    lastContent = content;
+    //TODO: Keeping for now for future reference:
+    //chrome.storage.sync.set({ content });
+  }
+};
+
+//Execute icon from text-select listener, trigger sidebar when clicked
+iconButton.onclick = (event) => {
+  event.stopPropagation();
+  iconButton.style.display = 'none';
+  createSidebar(lastContent);
+};
+
+//Exclude sidebar from text-select listener
+sidebar.onclick = (event) => {
+  event.stopPropagation();
+};
+
+function showIcon(event, content) {
+  iconButton.style.display = 'block';
+  iconButton.style.left = event.pageX + 5 + 'px';
+  iconButton.style.top = event.pageY - 45 + 'px';
+}
+
+function closeSidebar() {
+  sidebar.style.display = 'none';
+}
+
+//Inject sidebar html content
+function createSidebar(content) {
+  sidebar.innerHTML = '';
+  sidebar.style.display = 'block';
+  const panelHeading = document.createElement('div');
+  panelHeading.classList.add('message-header');
+  panelHeading.innerText = 'Lobster Ice Cream';
+  //TODO: Include icon in panelhead, update looks
+
+  const close = document.createElement('a');
+  close.classList.add('delete');
+  close.onclick = closeSidebar;
+  panelHeading.appendChild(close);
+  sidebar.appendChild(panelHeading);
+
+  const form = document.createElement('form');
+  const highlightLabel = document.createElement('label');
+  highlightLabel.classList.add('label', 'mt-2');
+  highlightLabel.innerText = 'Highlighted Text';
+  form.appendChild(highlightLabel);
+
+  const highlightTextarea = document.createElement('textarea');
+  highlightTextarea.classList.add('textarea');
+  form.appendChild(highlightTextarea);
+  highlightTextarea.value = content;
+
+  const noteTextarea = document.createElement('textarea');
+  noteTextarea.classList.add('textarea', 'mt-6');
+  noteTextarea.setAttribute('placeholder', 'Note to self');
+  form.appendChild(noteTextarea);
+
+  const buttonContainer = document.createElement('div');
+  buttonContainer.classList.add('has-text-centered', 'mt-1');
+  const addButton = document.createElement('button');
+  addButton.classList.add('button', 'is-link');
+  addButton.innerHTML = 'Add Clipping';
+  buttonContainer.appendChild(addButton);
+  form.appendChild(buttonContainer);
+
+  sidebar.appendChild(form);
+}
```

Comment:
Not sure if this is the best way to make this modular but I was think that you could have a function that takes in the kind of element you want to include, the input(s) to classList.add (either as an array or empty string the second input if second input not required?), and inner text (empty string if none). Then this function will create the element, does the classList.add, and also sets the innerHTML.

Reply:
okay i will try!